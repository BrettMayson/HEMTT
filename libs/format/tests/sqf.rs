use hemtt_format::{FormatterConfig, format_sqf};

const ROOT: &str = "tests/sqf/";

macro_rules! sqf {
    ($dir:ident) => {
        paste::paste! {
            #[test]
            fn [<sqf_ $dir>]() {
                insta::assert_snapshot!(sqf(stringify!($dir)));
            }
        }
    };
}

sqf!(arrays);
sqf!(blocks);
sqf!(cba);
sqf!(comments);
sqf!(format);
sqf!(hash);
sqf!(if_and);
sqf!(if_not);
sqf!(if);
sqf!(include);
sqf!(macro_path);
sqf!(macros);
sqf!(numbers);
sqf!(preserve_lines);
sqf!(private);
sqf!(sameline);

fn sqf(file: &str) -> String {
    format_sqf(
        &std::fs::read_to_string(format!("{ROOT}{file}.sqf")).expect("Failed to read test file"),
        &FormatterConfig::default(),
    )
    .unwrap_or_else(|err| format!("ERROR: {err}"))
}
