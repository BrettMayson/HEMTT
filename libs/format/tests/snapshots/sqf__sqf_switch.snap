---
source: libs/format/tests/sqf.rs
expression: sqf(stringify! (switch))
---
switch (floor random 6) do {
    case 1;
    case 3;
    case 5: { systemChat "Odd case selected"; };
    case 0;
    case 2;
    case 4: { systemChat "Even case selected"; };
    default { systemChat "Default case selected"; };
};
